### ISSUE TITLE
Defer selectPicker update until dropdown is closed

### ISSUE THREAD

**Issue by chrisknoll** [dreamRs/shinyWidgets#372]
I would like to know if it is possible to defer notifying the R server of the selection changes in a multiSelect until the dropdown is closed.

In our application, we present the user with filter options that will trigger a database query when the options are changed.  The problem is that as you go through making the choices, the server gets notified with the new set of options (while the dropdown is open) and causes a lot of needless traffic on our database.  It would be awesome if there was a `defer` option to the select picker such that it only notifies the R Server with the new choices after the dropdown is closed.  There is a javascript event that the bootstrap component signals when the dropdown is closed, but I am not good enough with Shiny + javascript to try to hook into the event.  It would be much simpler if there was an option in the UI component that would just defer updates until the select dropdown is closed.

**Comment by pvictor** [comment]
Hello,

With last version (CRAN or GitHub), you can try : 

````r
library(shiny)
library(shinyWidgets)

ui <- fluidPage(
  fluidRow(
    column(
      width = 6,
      pickerInput(
        inputId = "picker",
        label = "Pick some months",
        choices = month.name,
        multiple = TRUE
      )
    ),
    column(
      width = 6,
      verbatimTextOutput("result")
    )
  )
)

server <- function(input, output, session) {
  
  rv <- reactiveValues(value = NULL)
  
  observeEvent(input$picker_open, {
    if (isFALSE(input$picker_open)) {
      rv$value <- input$picker
    }
  })
  
  output$result <- renderPrint({
    rv$value
  })
  
}

shinyApp(ui, server)
```

`input$<inputId>_open` is a special input worth `TRUE` if menu is opened, or `FALSE` if closed.
Tell me if it's good enough for you, otherwise I'll look for the defer option.

Victor

**Comment by chrisknoll** [comment]
I'll give this a shot and let you know if it works.  Thank you!

**Comment by chrisknoll** [comment]
Ok, the solution you added works for us. Thanks very much!

Closed in c3d9758e5ce97622d25d1207444def76a4f4f281.
